File: algorithms/Array/go/1-TwoSum/two_sum.go
func twoSum(nums []int, target int) []int {
    res := make(map[int]int)
    for i:=0; i< len(nums); i++ {
        if idx, ok := res[target-nums[i]]; ok{
            return []int{idx, i}
        }
        res[nums[i]] = i
    }
    return nil
}

File: algorithms/Array/1-TwoSum/py/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = {}
        for idx, value in enumerate(nums):
            another = target - value
            if another in res:
                return [res[another], idx]
            else:
                res[value] = idx
        return None
